Utils/github_workflow_scripts/autobump_rn.py

```diff
@@ -0,0 +1,1168 @@
+        ), "Condition that should be skipped, should have a reason."
+        self.should_skip = should_skip
+        self.reason = reason
+        self.conflicting_packs = conflicting_packs
+        self.pack_new_rn_file = pack_new_rn_file
+        self.pr_rn_version = pr_rn_version
+        self.update_type = update_type
+
+    def __add__(self, other: "ConditionResult"):
+        """Sum of conditional results. Contains data of other conditional result if not empty, else self result.
+        Args:
+            other: conditional result to sum with.
+        Returns:
+            New conditional result, sum of both.
+        """
+        should_skip = self.should_skip or other.should_skip
+        reason = self.reason or other.reason
+        conflicting_packs = other.conflicting_packs or self.conflicting_packs
+        pack_new_rn_file = other.pack_new_rn_file or self.pack_new_rn_file
+        pr_rn_version = other.pr_rn_version or self.pr_rn_version
+        update_type = other.update_type or self.update_type
+        return ConditionResult(
+            should_skip=should_skip,
+            reason=reason,
+            conflicting_packs=conflicting_packs,
+            pack_new_rn_file=pack_new_rn_file,
+            pr_rn_version=pr_rn_version,
+            update_type=update_type,
+        )
+
+
+class BaseCondition(ABC):
+    """Base abstract class for conditions"""
+
+    def __init__(self, pr: PullRequest, git_repo: Repo, **kwargs):
+        self.pr = pr
+        self.git_repo = git_repo
+        self.next_cond = None
+
+    def set_next_condition(self, condition: "BaseCondition") -> "BaseCondition":
+        """
+        Args:
+            condition: next condition to check after current condition.
+        Returns:
+            next condition to check after current condition.
+        """
+        self.next_cond = condition  # type: ignore[assignment]
+        return self.next_cond   # type: ignore[return-value]
+
+    @abstractmethod
+    def generate_skip_reason(self, **kwargs) -> str:
+        """Abstract method. Will be over-written by classes that implements Condition."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def _check(
+        self, previous_result: Optional[ConditionResult] = None
+    ) -> ConditionResult:
+        """Abstract method. Will be over-written by classes that implements Condition."""
+        raise NotImplementedError
+
+    def check(
+        self, previous_result: Optional[ConditionResult] = None
+    ) -> ConditionResult:
+        """Checks conditions one after another.
+        Checks condition and if it is pass, checks next_condition.
+        If condition fails (should_skip set to True), exit and return last condition result.
+        Args:
+            previous_result: result of the previous condition that was handled.
+        Returns:
+            last checked condition's result.
+        """
+        curr_result = self._check(previous_result=previous_result)
```

Comment:
the following suggestion is totally optional, but it can reduce some repetitive code lines:
```suggestion
        self.previous_result = previous_result
        curr_result = ConditionResult(
            should_skip=self.should_skip(),
            reason=self.generate_skip_reason(),
            previous_result=self.previous_result,
        )
```
notes:
- `should_skip()` is basically `_check()` but instead of returning the `ConditionResult` obj, it returns the boolean `should_skip` value. Since `previous_result` is a class attribute, no need to pass it as an argument.
   * This is the most significant reason for my suggestion, you won't need to do `if ... return ConditionResult(...) else return ConditionResult(...)` in every `_check()` impl.
- `generate_skip_reason()` originally accepts arguments, but you can use class attributes and avoid that.
- similar for the rest of the class attributes that are currently passed to the `ConditionResult` c'tor like `conflicting_packs`, they can be derived from the previous_result.

Reply:
I didn't understand how to pass the previous_result (that I use like artifacts that passes from condition to condition), to the next condition? 
I mean where do I set the artifacts to be the the part of the previous result?

Reply:
let's talk about this